web/assets/js/main_out.js

```diff
@@ -1231,106 +1291,117 @@
+                pressed.q = true;
+                break;
+            case 69: // E
+                if (isTyping || escOverlayShown || pressed.e) break;
+                wsSend(UINT8_CACHE[22]);
+                pressed.e = true;
+                break;
+            case 82: // R
+                if (isTyping || escOverlayShown || pressed.r) break;
+                wsSend(UINT8_CACHE[23]);
+                pressed.r = true;
+                break;
+            case 84: // T
+                if (isTyping || escOverlayShown || pressed.t) break;
+                wsSend(UINT8_CACHE[24]);
+                pressed.t = true;
+                break;
+            case 80: // P
+                if (isTyping || escOverlayShown || pressed.p) break;
+                wsSend(UINT8_CACHE[25]);
+                pressed.p = true;
+                break;
+            case 27: // esc
+                if (pressed.esc) break;
+                pressed.esc = true;
+                if (escOverlayShown) hideESCOverlay();
+                else showESCOverlay();
+                break;
+        }
+    }
+    function keyup(event) {
+        switch (event.keyCode) {
+            case 32: // space
+                pressed.space = false;
+                break;
+            case 87: // W
+                pressed.w = false;
+                break;
+            case 81: // Q
+                if (pressed.q) wsSend(UINT8_CACHE[19]);
+                pressed.q = false;
+                break;
+            case 69: // E
+                pressed.e = false;
+                break;
+            case 82: // R
+                pressed.r = false;
+                break;
+            case 84: // T
+                pressed.t = false;
+                break;
+            case 80: // P
+                pressed.p = false;
+                break;
+            case 27: // esc
+                pressed.esc = false;
+                break;
+        }
+    }
+    function handleScroll(event) {
+        if (event.target !== mainCanvas) return;
+        mouseZ *= event.deltaY > 0 ? 0.8 : 1.2;
+        mouseZ = Math.max(mouseZ, settings.moreZoom ? 0.08 : 1);
+        mouseZ = Math.min(mouseZ, 4);
+    }
 
     function init() {
         mainCanvas = document.getElementById("canvas");
         mainCtx = mainCanvas.getContext("2d");
-        chatBox = document.getElementById("chat_textbox");
-        volumeInput = document.getElementById("volumeInput");
+        chatBox = byId("chat_textbox");
+        soundsVolume = byId("soundsVolume");
         mainCanvas.focus();
-        function handleScroll(event) {
-            mouseZ *= Math.pow(.9, event.wheelDelta / -120 || event.detail || 0);
-            0.2 > mouseZ && (mouseZ = 0.2);
-            mouseZ > 4 / mouseZ && (mouseZ = 4 / mouseZ);
-        }
-        if (/firefox/i.test(navigator.userAgent))
-            document.addEventListener("DOMMouseScroll", handleScroll, false);
-        else
-            document.body.onmousewheel = handleScroll;
```

Comment:
The wheel event should be used, read the warning here https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/GlobalEventHandlers.onmousewheel